commit 578e60102871d11ed8c18d36f6286f3a96258d8f
Author: Ryan Goldberg <rgoldber@redhat.com>
Date:   Thu Dec 1 16:15:44 2022 -0500

    PR29676: Wildcard expansion fix for labels
    
    PR29676, introduced an bug where function symbols from the symbol
    table were expanded in the function component resulting in wildcards
    not being expanded in labels. This fix, removes the issue by restricting
    the symbol table query to probes which don't need further debuginfo to
    expand.

diff --git a/tapsets.cxx b/tapsets.cxx
index 0ec71ebda..46b10f26e 100644
--- a/tapsets.cxx
+++ b/tapsets.cxx
@@ -1256,7 +1256,11 @@ dwarf_query::handle_query_module()
   // in the symbol table but not in dwarf and minidebuginfo is
   // located in the gnu_debugdata section, alias_dupes checking
   // is done before adding any probe points
-  if(!pending_interrupts)
+  // PR29676.   Some probes require additional debuginfo
+  // to expand wildcards (ex. .label, .callee). Since the debuginfo is
+  // not available, don't bother looking in the symbol table for these results.
+  // This can result in 0 results, if there is no dwarf info present
+  if(!pending_interrupts && !(has_label || has_callee || has_callees_num))
     query_module_symtab();
 }
 

